### PR TITLE
[Exporter Refactor] Dev Branch

### PR DESCRIPTION
**Pull request by dbogunowicz**
Adds:
- `convert_quantizable_matmul` transform

### CODE REVIEW

File: src/sparseml/exporters/transforms/convert_quantizable_matmul.py [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+        for idx, matmul_node in zip(matmul_nodes_indices, matmul_nodes):
+            # Check whether there is a match
+            result = is_quantizable_matmul(matmul_node, graph)
+            if result is None:
+                continue
+            output_quantize_node, last_node_optional = result
+            _LOGGER.debug(f"Matched quantizable MatMul: {matmul_node.name}")
+
+            # Convert
+            model = convert_matmul_to_quantized(
```
Comment (dbogunowicz) [inline src/sparseml/exporters/transforms/convert_quantizable_matmul.py:292]: @corey-nm this is the particular implementation of what you described as `transform_match` method

File: src/sparseml/exporters/transforms/convert_quantizable_matmul.py [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+            )
+        return model
+
+    def _validate_input(self, model: ModelProto):
+        validate_onnx_file(model)
+
+    def _validate_output(self, model: ModelProto):
+        validate_onnx_file(model)
+
+    def __call__(self, model: Union[ModelProto, str]) -> ModelProto:
```
Comment (dbogunowicz) [inline src/sparseml/exporters/transforms/convert_quantizable_matmul.py:310]:
```suggestion
    def apply(self, model: Union[ModelProto, str]) -> ModelProto:
```

**Comment by dbogunowicz** [comment]
Let's not be too harsh when reviewing this, I am in the process of implementing another pass, and I am already seeing a lot of room for clever helper functions to minimize the amount of boilerplate. Let's probably schedule a mini PR party after we have implemented the set of most bloated and complicated passes to agree on the interfaces.

File: src/sparseml/exporters/transforms/convert_quantizable_conv_integer.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    This MatMul is the result of quantizing native torch.matmul using QATMatMul
+
+    | Starting with:
+    |          INPUT         QuantizeLinear (with constant kernel)
+    |            |               |
+    |     QuantizeLinear     DequantizeLinear
+    |            |               |
+    |     DequantizeLinear      |
+    |                  |      |
+    |                   Conv (with bias)
```
Comment (corey-nm) [inline src/sparseml/exporters/transforms/convert_quantizable_conv_integer.py:85]: idea
```python
match_nodes("ConvNode", parents=[
    ["DequantizeLinear", "QuantizeLinear"],
    ["DequantizeLinear", "QuantizeLinear"],
])
```

File: tests/sparseml/exporters/transforms/test_convert_quantizable_matmul.py [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+
+@pytest.mark.parametrize(
+    "with_transpose, with_reshape, testing_function",
+    [
+        (False, False, _test_matmul),
+        (True, False, _test_transpose),
+        (False, True, _test_reshape),
+        (True, True, _test_reshape_and_transpose),
+    ],
+)
```
Comment (corey-nm) [inline tests/sparseml/exporters/transforms/test_convert_quantizable_matmul.py:235]: 🔥

File: src/sparseml/exporters/transforms/convert_quantizable_conv_integer.py [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+                for idx, node in enumerate(model.graph.node)
+                if node.op_type == "Conv"
+            ]
+        )
+        for idx, conv_node in zip(conv_nodes_indices, conv_nodes):
+            # Check whether there is a match
+            if not is_quantizable_conv(conv_node, model):
+                continue
+            _LOGGER.debug(f"Matched quantizable Conv weight and bias: {conv_node.name}")
+
```
Comment (corey-nm) [inline src/sparseml/exporters/transforms/convert_quantizable_conv_integer.py:118]: Oh just noticed, is this function missing the transformation logic?

**Pull request closed without merging**